docs/proposals/20240807-in-place-updates.md

```diff
@@ -136,36 +132,36 @@ Cluster API user experience MUST be the same when using default, immutable updat
 If external update extensions can not cover the totality of the desired changes, CAPI WILL defer to Cluster API’s default, immutable rollouts. This is important for a couple of reasons:
 
 * It allows to implement custom rollout strategies incrementally, without the need to cover all use cases up-front.
-* There are case when replacing the machine will always be necessary:
+* There are cases when replacing the machine will always be necessary:
     * When it is not possible to recover the machine, e.g. hardware failure.
     * When the user determines that recovering the machine is too complex/costly vs replacing it. 
     * Automatic machine remediation (unless you use external remediation strategies)
 
 #### Clean separation of concern
 
-The external update extension will be responsible to perform the updates on a single machine.
+It is the responsibility of the extension to decide if it can perform changes in-place and to perform these changes on a single machine. If the extension decides that it cannot perform changes in-place, CAPI will fall back to rollout.
 
-The responsibility to determine which machine should be rolled out as well as the responsibility to handle rollout options like MaxSurge/MaxUnavailable will remain on the controllers owning the machine (e.g. KCP, MD controller).
+The responsibility to determine which machine should be rolled out as well as the responsibility to handle rollout options like MaxSurge/MaxUnavailable will remain on the controllers owning the machine (e.g. KCP, MD controller). 
 
 ### Goals
 
-- Enable the implementation of in-place update strategies.
+- Enable the implementation of pluggable update extensions.
 - Allow users to update Kubernetes clusters using pluggable External Update Extension.
 - Maintain a coherent user experience for both rolling and in-place updates.
 - Support External Update Extensions for both Control Plane (KCP or others) and MachineDeployment controlled machines.
-- Allow in-place updates for single-node clusters without the requirement to reprovision hosts.
 
 ### Non-Goals/Future work
 
 - To provide rollbacks in case of an in-place update failure. Failed updates need to be fixed manually by the user on the machine or by replacing the machine.
 - Introduce any changes to KCP (or any other control plane provider), MachineDeployment, MachineSet, Machine APIs.
 - Ammend the desired state to something that the registered updaters can cover or register additional updaters capable of handling the desired changes.
```

Comment:
There was a comment about this sentence being hard to understand, maybe we can remove it?

Reply:
Moved it to `Non-Goals/Future work` section as per [suggestion](https://github.com/kubernetes-sigs/cluster-api/pull/11029/files#r2008927060) but I think it's fine to remove it.